homeassistant/components/squeezebox/media_player.py

```diff
@@ -182,133 +180,52 @@ async def async_service_handler(service):
-        )
-        url = f"http://{self.host}:{self.port}/jsonrpc.js"
-        data = json.dumps(
-            {"id": "1", "method": "slim.request", "params": [player, command]}
-        )
-
-        _LOGGER.debug("URL: %s Data: %s", url, data)
-
-        try:
-            websession = async_get_clientsession(self.hass)
-            with async_timeout.timeout(TIMEOUT):
-                response = await websession.post(url, data=data, auth=auth)
-
-                if response.status != HTTP_OK:
-                    _LOGGER.error(
-                        "Query failed, response code: %s Full message: %s",
-                        response.status,
-                        response,
-                    )
-                    return False
-
-                data = await response.json()
-
-        except (asyncio.TimeoutError, aiohttp.ClientError) as error:
-            _LOGGER.error("Failed communicating with LMS: %s", type(error))
-            return False
-
-        try:
-            return data["result"]
-        except AttributeError:
-            _LOGGER.error("Received invalid response: %s", data)
-            return False
-
-
 class SqueezeBoxDevice(MediaPlayerEntity):
-    """Representation of a SqueezeBox device."""
+    """
+    Representation of a SqueezeBox device.
 
-    def __init__(self, lms, player_id, name):
+    Wraps a pysqueezebox.Player() object.
+    """
+
+    def __init__(self, hass, player):
         """Initialize the SqueezeBox device."""
         super().__init__()
-        self._lms = lms
-        self._id = player_id
-        self._status = {}
-        self._name = name
+        self._hass = hass
+        self._player = player
         self._last_update = None
-        _LOGGER.debug("Creating SqueezeBox object: %s, %s", name, player_id)
+        _LOGGER.debug(
+            "Creating SqueezeBox object: %s, %s", player.name, player.player_id
+        )
 
     @property
     def name(self):
         """Return the name of the device."""
-        return self._name
+        return self._player.name
 
     @property
     def unique_id(self):
         """Return a unique ID."""
-        return self._id
+        return self._player.player_id
 
     @property
     def state(self):
         """Return the state of the device."""
-        if "power" in self._status and self._status["power"] == 0:
-            return STATE_OFF
-        if "mode" in self._status:
-            if self._status["mode"] == "pause":
+        if self._player.power is not None:
+            if not self._player.power:
+                return STATE_OFF
+        if self._player.mode:
+            if self._player.mode == "pause":
                 return STATE_PAUSED
-            if self._status["mode"] == "play":
+            if self._player.mode == "play":
                 return STATE_PLAYING
-            if self._status["mode"] == "stop":
+            if self._player.mode == "stop":
                 return STATE_IDLE
```

Comment:
Please make a constant dict at the module level that maps between player mode and home assistant state. Then look up that dict here.

Reply:
Done. Added to new commit.